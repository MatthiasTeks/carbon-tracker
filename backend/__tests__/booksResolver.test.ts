import { execute } from '../jest.setup';
import Book from '../src/entities/book';
import getBooksQuery from './operations/getBooks';

describe('Books resolver', () => {
  it('can get a list of books', async () => {
    await Book.create({ title: 'title1', author: 'author1' }).save();
    await Book.create({ title: 'title2', author: 'author2' }).save();
    const res = await execute(getBooksQuery);
    expect(res).toMatchSnapshot();
  });
});
