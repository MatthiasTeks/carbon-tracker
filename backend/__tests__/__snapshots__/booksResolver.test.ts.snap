// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books resolver can get a list of books 1`] = `
{
  "data": {
    "tags": [
      {
        "author": "author2",
        "id": 2,
        "title": "title2",
      },
      {
        "author": "author1",
        "id": 1,
        "title": "title1",
      },
    ],
  },
}
`;
